crates/matrix-sdk-crypto/src/backups/keys/decryption.rs

```diff
@@ -298,13 +334,102 @@ mod tests {
         let key_backup_data: KeyBackupData = serde_json::from_value(data).unwrap();
         let ephemeral = key_backup_data.session_data.ephemeral.encode();
         let ciphertext = key_backup_data.session_data.ciphertext.encode();
-        let mac = key_backup_data.session_data.mac.encode();
+        let mac = key_backup_data.session_data.mac.unwrap().encode();
 
         let decrypted = decryption_key
-            .decrypt_v1(&ephemeral, &mac, &ciphertext)
+            .decrypt_v1(&ephemeral, Mac::V1(&mac), &ciphertext)
             .expect("The backed up key should be decrypted successfully");
 
         let _: BackedUpRoomKey = serde_json::from_str(&decrypted)
             .expect("The decrypted payload should contain valid JSON");
     }
+
+    #[test]
+    fn test_mac2_check() {
+        let decryption_key =
+            BackupDecryptionKey::from_base64("Ha9cklU/9NqFo9WKdVfGzmqUL/9wlkdxfEitbSIPVXw")
+                .unwrap();
+
+        // mac2 was created from the data in test_decrypt_key using the following Python:
```

Comment:
I think you should be able to use markdown for this and even specify the type:

```python
import ...
```

This will likely even syntax highlight the code in some smarter editors.